drivers/hmis/app/jobs/hmis/create_fake_enrollments_job.rb

```diff
@@ -0,0 +1,311 @@
+
+# == Hmis::CreateFakeEnrollmentsJob
+#
+# Tool to generate fake Enrollment data for local development and staging environments.
+# It generates households of varying sizes. The default behavior is to generate open enrollments for existing clients.
+#
+# Records it generates:
+# - Clients (optionally)
+# - Enrollments (Household size varies from 1-4, except for NBN shelters which are set to 1)
+# - Exits (optionally)
+# - Disabilities (only for intake stage)
+# - Bed Night Services (for NBN projects, 1-100 per Enrollment)
+# - Intake and Exit Assessments
+#
+# Usage:
+#   Hmis::CreateFakeEnrollmentsJob.perform_now(num_households: 10)
+#   Hmis::CreateFakeEnrollmentsJob.perform_now(num_households: 10, generate_clients: true, exited: true, project_ids: [72])
+module Hmis
+  class CreateFakeEnrollmentsJob < BaseJob
+    include Hmis::Concerns::HmisArelHelper
+    include NotifierConfig
+    queue_as ENV.fetch('DJ_LONG_QUEUE_NAME', :long_running)
+
+    BATCH_SIZE = 5_000
+    HOUSEHOLD_SIZE_RANGE = 1..4
+    ENTRY_DATE_RANGE = 1..(365 * 3) # 1d-3y ago
+
+    def perform(
+      num_households:,
+      generate_clients: false,
+      exited: false,
+      project_ids: nil,
+      data_source_id: nil,
+      export_id: 'FAKED'
+    )
+      raise "won't create fake enrollments in production" if Rails.env.production?
+
+      setup_notifier('CreateFakeEnrollmentsJob')
+      @data_source = if data_source_id
+        ::GrdaWarehouse::DataSource.hmis.find(data_source_id)
+      else
+        ::GrdaWarehouse::DataSource.hmis.sole # 'sole' will raise if there are >1 HMIS data sources
+      end
+
+      @hud_user_id = Hmis::Hud::User.system_user(data_source_id: @data_source.id).user_id
+      @faker = ::GrdaWarehouse::FakeData.new
+      @export_id = export_id
+
+      project_scope = if project_ids
+        Hmis::Hud::Project.where(data_source: @data_source, id: project_ids)
+      else
+        Hmis::Hud::Project.where(data_source: @data_source)
+      end
+
+      # Generate and import data in batches
+      remaining = num_households
+      num_batches = (num_households / BATCH_SIZE) + 1
+      num_batches.times do |i|
+        n = i + 1 == num_batches ? remaining : BATCH_SIZE
+        log("Generating #{n} #{exited ? 'exited' : 'active'} households across #{project_scope.size} projects, #{generate_clients ? 'for new clients' : 'for existing clients'} (batch #{i + 1}/#{num_batches})...")
+        Hmis::Hud::Base.transaction do
+          generate_and_import_enrollment_data(
+            num_households: n,
+            generate_clients: generate_clients,
+            exited: exited,
+            projects: project_scope,
+          )
+        end
+        remaining -= n
+      end
+
+      # queue client deduplication and service history processing
+      Hmis::Hud::Client.last.send(:warehouse_identify_duplicate_clients) if generate_clients && !Rails.env.development?
+      Hmis::Hud::Enrollment.last.send(:queue_service_history_processing!)
```

Comment:
Ah thanks, yeah that was a bit hacky. I updated it to use class methods as you suggested. I also removed the the dev check, because it's fine to run `warehouse_identify_duplicate_clients` in dev, it just seems to run in place instead of queuing (in development)